### PR TITLE
wagtail 3.0 path updates

### PR DESCRIPTION
**Pull request by William-Blackie**
Hey all,

This PR adds support for the new Wagtail 3.0+ path as well as backward compatibility for wagtail <=2 paths.

I've followed the official docs found [here](https://docs.wagtail.org/en/stable/releases/3.0.html#upgrade-considerations-changes-affecting-all-projects). 

Let me know if this works for you.

Thanks - Will. 
```
./runtests.py
Found 73 test(s).
Creating test database for alias 'default'...
Adding content type 'wagtail_transfer | idmapping'
Adding content type 'wagtail_transfer | importedfile'
System check identified no issues (0 silenced).
.../home/william/.local/lib/python3.8/site-packages/rest_framework/pagination.py:200: UnorderedObjectListWarning: Pagination may yield inconsistent results with an unordered object_list: <class 'tests.models.Category'> QuerySet.
  paginator = self.django_paginator_class(queryset, page_size)
......................................................................
----------------------------------------------------------------------
Ran 73 tests in 5.127s

OK

```

### CODE REVIEW

**Review by BrianXu20 — COMMENTED** [review]
The changes look good, 
however,  I think you should modify [config.yml](https://github.com/wagtail/wagtail-transfer/blob/main/.circleci/config.yml) to include wagtail 3.0 tests. 
Probably keep the tests between Wagtail 2.15 - 3.0

**Review by BrianXu20 — APPROVED** [review]

File: CHANGELOG.txt [hunk 1]
```diff
@@ -1,5 +1,9 @@
 Changelog
 =========
+0.9.0 (16.06.2022)
```
[the remark this answers is not in the document]
Comment (William-Blackie) [reply CHANGELOG.txt:3]: Okay, thank you for the review @gasman.

**Comment by William-Blackie** [comment]
I've made the above requested changes.

**Review by gasman — APPROVED** [review]
Looks good to me, thanks!

**Pull request merged**